ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -295,6 +300,26 @@ private Response executeStatements(
     }
   }
 
+  private void waitForPreviousDistributedStatementToBeHandled(final KsqlEntityList entities) {
+    final ArrayList<KsqlEntity> reversed = new ArrayList<>(entities);
+    Collections.reverse(reversed);
+
+    reversed.stream()
+        .filter(e -> e instanceof CommandStatusEntity)
+        .map(cs -> ((CommandStatusEntity) cs).getCommandSequenceNumber())
+        .filter(Objects::nonNull)
+        .findFirst()
+        .ifPresent(seqNum -> {
+          try {
+            commandQueue.ensureConsumedPast(seqNum, distributedCmdResponseTimeout);
+          } catch (final InterruptedException e) {
+            throw new KsqlRestException(Errors.serverShuttingDown());
```

Comment:
For my own understanding: why is it the case that any `InterruptedException` caught here would be caused by a server shutdown?

Reply:
Well... InterruptedException isn't _expected_ with the current code base, but because its a checked exception we have to handle it. (I hate checked exceptions!).  So strictly speaking its not currently shutting down. However, the only time we're likely to get  `InterruptedException` in the future is if someone adds shutdown code to the server which interrupts all running threads etc.